indico/queries/teach_tasks.py

```diff
@@ -0,0 +1,244 @@
+                "Examples not found. Please check the ID you are using."
+            )
+        return examples
+
+
+class CreateQuestionaire(GraphQLRequest):
+    """
+    Creates the questionnaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        dataset_id (int): The id of the dataset to create the questionnaire from.
+        source_column_id (int): The id of the source column to create a questionnaire from.
+        targets (List[str]): The classes for the dataset.
+
+    Returns:
+        Questionnaire
+
+    """
+
+    query = """
+        mutation(
+            $name: String!,
+            $dataset_id: Int!,
+            $questions: [QuestionInput]!,
+            $source_col_id: Int!,
+        ) {
+            createQuestionnaire (
+                datasetId: $dataset_id,
+                dataType: TEXT,
+                name: $name,
+                numLabelersRequired: 1,
+                questions: $questions,
+                sourceColumnId: $source_col_id,
+                instructions: ""
+            ) {
+                id
+            }
+        }
+    """
+
+    def __init__(
+        self, name: str, dataset_id: int, source_column_id: int, targets: List[str]
+    ):
+        questions = [
+            {"type": "ANNOTATION", "targets": targets, "keywords": [], "text": name,}
+        ]
+        super().__init__(
+            query=self.query,
+            variables={
+                "name": name,
+                "dataset_id": dataset_id,
+                "questions": questions,
+                "source_col_id": source_column_id,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            questionnaire = Questionnaire(
+                **super().process_response(response)["createQuestionnaire"]
+            )
+        except IndexError:
+            raise IndicoNotFound("Failed to create Questionnaire")
+        return questionnaire
+
+
+class CreateTeachTask(RequestChain):
+    """
+    Creates a labeled questionaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        csv_path (str): Path to a csv with columns containing text and json encoded labels.
+        dataset (Dataset): Dataset to create the questionnaire from.
+        num_examples (int): number of rows in the dataset in total.
+        text_column (str): Optional. The column name in the csv containing the text.
+        label_column (str): Optional. The column name in the csv containing the json encoded labels. 
+        
+    Returns:
+        Dataset object   
+    """
+
+    previous = None
+
+    def __init__(
+        self,
+        name: str,
+        csv_path: str,
+        dataset: Dataset,
+        num_examples: int,
+        text_column="text",
+        label_column="labels",
+    ):
+        self.dataset_id = dataset.id
+        self.dataset = dataset
+        csv = pd.read_csv(csv_path)
```

Comment:
I'm not a fan of the interface here, is there a reason we require a csv for this? It's limiting in context where the user doesn't have a file.